### ISSUE TITLE
Support for specifying public IP for Azure API Management

### ISSUE THREAD

**Issue by sikovatc** [hashicorp/terraform-provider-azurerm#12021]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I tried to edit the public IP for an Azure API Management Instance through Terraform, and it looks like the public IP is not an editable field currently. I see it’s available as an output variable, but not an input variable. 
I tried a few times to set it as shown below. 

This was an Azure APIM instance that I had previously created so was trying to edit the public IP setting for it; currently, you can edit this in the Azure portal, PowerShell, CLI, etc. 

The error I received was: "This field cannot be set"

### New or Affected Resource(s)

* azurerm_api_management

### Potential Terraform Configuration

```hcl
virtual_network_type = "External"

virtual_network_configuration {
subnet_id = data.azurerm_subnet.example.id
}
additional_location {
location = "East US"
public_ip_addresses = var.public_ips
}
```

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/api_management

* #0000

**Comment by yupwei68** [comment]
Hi @sikovatc , thanks for opening this issue.
According to the document https://docs.microsoft.com/en-us/azure/api-management/api-management-howto-ip-addresses , I think ip address of api management service is not settable.
Would you provide portal, cli or powershell cmd to set the ip address or ip addresses of api management service?

**Comment by sikovatc** [comment]
Hi Yuping,

Yes, you can change the pubic IP of an existing APIM instance:


***@***.***D75D12.F87B3EA0]

Thanks,

Simona Kovatcheva
Sr. Cloud Solution Architect      Cell: (480)388-5756
US National Team|                       ***@***.***
***@***.***D75D12.F87B3EA0]

**Comment by sikovatc** [comment]
HI @yupwei68 you can change the public IP of an already created APIM instance:
![apim](https://user-images.githubusercontent.com/53770775/121375928-b8106100-c906-11eb-974a-651f43789a1a.png)

**Comment by tsubasaxZZZ** [comment]
Hi @yupwei68,
I agree with this Issue. I would like to set zones, but zones requests public IP address. So I need set public IP.
In this document, it seem to be able to set public IP when create.
https://docs.microsoft.com/en-us/rest/api/apimanagement/2021-01-01-preview/api-management-service/create-or-update

**Comment by thomast1906** [comment]
+1 to this, we want to deploy `azurerm_api_management` using Availability Zones but currently Terraform does not support this.

Error below: 
> Error: creating/updating API Management Service "tamops-api-mgmt-sbox-prem" (Resource Group "tamops-apim-tf"): apimanagement.ServiceClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="NotSupported" Message="Deploying Microsoft.ApiManagement/service into Virtual Network in location `UK South` and configuring Availability Zones (1,2,3) is only supported in api-version 2021-01-01-preview,2021-04-01-preview and setting property `publicIpAddressId` is a must. Refer to aka.ms/azurestandardpublicip on how to create a Standard SKU Public IP Address resource."

**Comment by mgpole** [comment]
+1 I have run into the same issue trying to deploy APIM internal on virtual network using availability zones. Get the following error

`Error: creating/updating API Management Service "apim" (Resource Group "apim-rg"): apimanagement.ServiceClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="NotSupported" Message="Deploying Microsoft.ApiManagement/service into Virtual Network in location `Australia East` and configuring Availability Zones (1,2) is only supported in api-version 2021-01-01-preview,2021-04-01-preview and setting property publicIpAddressId is a must. Refer to aka.ms/azurestandardpublicip on how to create a Standard SKU Public IP Address resource."`

**Comment by timja** [comment]
This should be unblocked once the SDK is upgraded, see https://github.com/Azure/azure-sdk-for-go/issues/15504#issuecomment-924535336

**Comment by thomast1906** [comment]
@yupwei68 - could this be reviewed against this? https://github.com/hashicorp/terraform-provider-azurerm/pull/12566 

We're looking the functionality as referenced above

**Comment by severity1** [comment]
Waiting for this.

**Comment by lrsvendsen** [comment]
PublicIPAddressID has not been implemented. It is blocking two core deployment options "addition locations and "multiple zones". See issue #13445

**Comment by oWretch** [comment]
I am actively working on implementing this as we require it for a project. Code is mostly done and I plan to have a PR submitted early next week.

**Comment by himeshbhojwani** [comment]
Please fix this issue

**Comment by thomast1906** [comment]
Still unsure if PublicIPAddressID  is implemented correctly; when attempting now I get error "Service activation failed. Please look at the details in Activity Log on the left side. In case you are deploying into VNET please make sure prerequisites are followed as described on https://aka.ms/apimvnet"

`publicIpAddressId` is not properties of json output.

**Comment by thomast1906** [comment]
@katbyte when you get time, can you confirm?

**Comment by oWretch** [comment]
@thomast1906 Can you open an issue for this? I'm happy to have a look at this if you can provide more specific details on how you get to the error.

**Comment by thomast1906** [comment]
@oWretch thanks, sorry I should have updated further.

Seems to be fixed in v2.97.0 rather than the mentioned v2.96.0

Have a successful APIM deployment now premium SKU and availability zones
